### PR TITLE
Add Architecture section in README.md

### LINKED ISSUE #34: Add some "architecture" documentation
- project structure description
- how to inject a class, how to mock it for testing
- where to add tests
- etc.

Should be useful to onboard a new developer, but also when we open the project.

### PR DESCRIPTION
**Pull request by hamorillo**
Closes #34 

Adding some architecture documentation to the Readme.

### CODE REVIEW

**Comment by maxme** [comment]
LGTM, but IMO we shouldn't merge this now
- Wait for the split and the big rename to update this doc (new module and updated names).
- Now that we generate models and the rest client via OpenAPI specs, we should add a note about it.

**Comment by hamorillo** [comment]
I think we can merge the Architecture section, so I've updated the doc. Let me know what you think.

File: README.md [hunk 1]
```diff
@@ -120,21 +162,24 @@ repositories {
...
+code. [Dokka](https://kotlinlang.org/docs/dokka-introduction.html) has been setup to generate html documentation from kdoc. To generate the
+HTML docs in `docs/dokka/`, run the following command:
 
 ```sh
 ./gradlew dokkaHtmlMultiModule
 ```
 
-Here is the [Latest Gravatar-SDK-Android API documentation generated by Dokka](https://automattic.github.io/Gravatar-SDK-Android/trunk/index.html)
+Here is
+the [Latest Gravatar-SDK-Android API documentation generated by Dokka](https://automattic.github.io/Gravatar-SDK-Android/trunk/index.html)
```
Comment (maxme) [inline README.md]: Super nitpicking, but if we go that route and limit line width on markdown docs, we should maybe switch to [reference-style links][1]

[1]: https://www.markdownguide.org/basic-syntax/#reference-style-links

**Review by maxme — APPROVED** [review]
:shipit:

File: README.md (hunk 1 above)
Comment (hamorillo) [reply README.md]: I didn't want to do that 🤦🏼‍♂️. I use the IDE formatting before commiting... my bad. 

Updated.

**Pull request merged**